xarray/tests/test_dataarray.py

```diff
@@ -2518,6 +2518,57 @@ def test_combine_first(self):
                              [('x', ['a', 'b', 'd']), ('y', [-1, 0])])
         self.assertDataArrayEqual(actual, expected)
 
+    def test_sortby(self):
+        da = DataArray([[1, 2], [3, 4], [5, 6]],
+                       [('x', ['c', 'b', 'a']), ('y', [1, 0])])
+
+        expected = DataArray([[5, 6], [3, 4], [1, 2]],
+                             [('x', ['a', 'b', 'c']), ('y', [1, 0])])
+
+        actual = da.sortby('x')
+        self.assertDataArrayEqual(actual, expected)
+
+        dax = DataArray([100, 90, 80], [('x', [0, 1, 2])])
+        actual = da.sortby(dax)
+        self.assertDataArrayEqual(actual, expected)
+
+        # test 1-D lexsort
+        # dax0 is sorted first to give indices of [1, 2, 0]
+        # and then dax1 would be used to move index 2 ahead of 1
+        dax0 = DataArray([100, 95, 95], [('x', [0, 1, 2])])
+        dax1 = DataArray([0, 1, 0], [('x', [7, 8, 9])])
```

Comment:
This example actually raises a good question, because `dax1` has different coordinate labels for `x` than `dax0` and `da`.

Should we align sortby arguments before using them, or ensure that they are already aligned? I think this is probably a good idea.